src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs

```diff
@@ -126,26 +126,35 @@ private void UsingDeclaration()
             // Set block type to directive
             Context.CurrentBlock.Type = BlockType.Directive;
 
-            // Parse a type name
-            Assert(CSharpSymbolType.Identifier);
-            NamespaceOrTypeName();
-            IEnumerable<CSharpSymbol> ws = ReadWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
-            if (At(CSharpSymbolType.Assign))
+            if (At(CSharpSymbolType.Identifier))
             {
-                // Alias
-                Accept(ws);
-                Assert(CSharpSymbolType.Assign);
-                AcceptAndMoveNext();
+                // non-static using
+                NamespaceOrTypeName();
+                IEnumerable<CSharpSymbol> ws = ReadWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
+                if (At(CSharpSymbolType.Assign))
+                {
+                    // Alias
+                    Accept(ws);
+                    Assert(CSharpSymbolType.Assign);
+                    AcceptAndMoveNext();
 
-                AcceptWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
+                    AcceptWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
 
-                // One more namespace or type name
-                NamespaceOrTypeName();
+                    // One more namespace or type name
+                    NamespaceOrTypeName();
+                }
+                else
+                {
+                    PutCurrentBack();
+                    PutBack(ws);
+                }
             }
-            else
+            else if (At(CSharpKeyword.Static))
             {
-                PutCurrentBack();
-                PutBack(ws);
+                // static using
+                AcceptAndMoveNext();
+                AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
```

Comment:
That's how our `@using` imports have always been. In general, Razor directives populate only a single line.

Not to mention having an errored using like so:

```HTML
@using 
Favorite Movie: <strong>Abc</strong>
```
Would be terribly confusing to the user because Razor would attempt to do: `using Favorite` at the top of the generated C# file.